File: Java/labs/labs12_Flyweight/task_2_2/src/Car.java
public class Car {
  public enum CarColor {
    White,
    Black,
    Red,
    Grey
  }

  public enum Type {
    Sedan,
    Hatchback,
    SUV
  }

  final private Type type;
  final private CarColor carColor;
  final private Engine engine;
  final private Wheel wheel;

  public Car(Type type, CarColor carColor,
             Engine engine, Wheel wheel) {
    this.type = type;
    this.carColor = carColor;
    this.engine = engine;
    this.wheel = wheel;
  }

  public void showInfo() {
    System.out.println( "Car:\n" +
            "type=" + type +
            ",\ncarColor=" + carColor +
            ",\nengine=" + engine +
            ",\nwheel=" + wheel +
            '\n');
  }
}

File: Cpp/lab04_Builder/task_3_1/src/car.cpp
#include "car.hpp"


Car::Car(Car::CarType type, Car::CarColor color,
Engine engine, Wheel wheel)
: type(type), color(color), engine(engine), wheel(wheel) {}

std::ostream& operator<<(std::ostream& os, Car const& car)
{
  os << "Car:\ntype=";

  switch (car.type)
  {
    case Car::CarType::Sedan:
      os << "Sedan";
      break;
    case Car::CarType::Hatchback:
      os << "Hatchback";
      break;
    case Car::CarType::SUV:
      os << "SUV";
      break;
  }

  os << ",\ncarColor=";

  switch (car.color)
  {
    case Car::CarColor::White:
      os << "White";
      break;
    case Car::CarColor::Black:
      os << "Black";
      break;
    case Car::CarColor::Red:
      os << "Red";
      break;
    case Car::CarColor::Grey:
      os << "Grey";
      break;
  }

  os << ",\nengine=" << car.engine
     << ",\nwheel=" << car.wheel;

  return os;
}
